homeassistant/components/group/sensor.py

```diff
@@ -330,9 +356,26 @@ def async_update_group_state(self) -> None:
                         self._state_incorrect.add(entity_id)
                         _LOGGER.warning(
                             "Unable to use state. Only numerical states are supported,"
-                            " entity %s with value %s excluded from calculation",
+                            " entity %s with value %s excluded from calculation in %s",
+                            entity_id,
+                            state.state,
+                            self.entity_id,
+                        )
+                    continue
+                except (KeyError, HomeAssistantError):
+                    valid_states.append(False)
+                    if entity_id not in self._state_incorrect:
+                        self._state_incorrect.add(entity_id)
+                        _LOGGER.warning(
+                            "Unable to use state. Only entities with correct unit of measurement"
+                            " is supported when having a device class,"
+                            " entity %s, value %s with device class %s"
+                            " and unit of measurement %s excluded from calculation in %s",
                             entity_id,
                             state.state,
+                            self.device_class,
+                            state.attributes.get("unit_of_measurement"),
+                            self.entity_id,
```

Comment:
Setting incompatible units will in the not too distant future prevent sensor entities from writing states. I think we should add a comment here saying the checks here can be simplified or removed when sensor blocks incorrect states.